velox/connectors/hive/HiveConnector.cpp

```diff
@@ -400,6 +400,45 @@ velox::variant convertFromString(const std::optional<std::string>& value) {
   return velox::variant(ToKind);
 }
 
+// A quick implementation to convert string as int128.
+int128_t atoint128_t(const char *s) {
+  const char *p = s;
+  int128_t val = 0;
+
+  if (*p == '-' || *p == '+') {
+    p++;
+  }
```

Comment:
This implement does not consider overflow, is it possible overflow? I will have a PR to cast varchar to decimal. https://github.com/oap-project/velox/pull/169/files#diff-c6d2f99b68772f908f2f3e8b979b3c2894f790f20af68684e7b0400bed969389R222 Can you use this one after it merge?

Reply:
This implement origins from velox https://github.com/facebookincubator/velox/blob/main/velox/type/Conversions.h#L89

Reply:
Sure. I'll take a look, thanks.

Reply:
@jinchengchenghh Updated. Can you help review again?